src/routes/accounts.py

```
@@ -1,25 +1,244 @@
+from security.token_manager import JWTAuthManager
 
 from config import get_jwt_auth_manager, get_settings, BaseAppSettings
+from crud import create_user, get_user_by_email
 from database import (
     get_db,
     UserModel,
     UserGroupModel,
     UserGroupEnum,
     ActivationTokenModel,
     PasswordResetTokenModel,
-    RefreshTokenModel
+    RefreshTokenModel,
 )
 from exceptions import BaseSecurityError
+from schemas.accounts import (
+    UserRead,
+    CreateUser,
+    TokenActivate,
+    TokenPasswordRefresh,
+    ResetPassword,
+    LoginSchema,
+    LoginResponseSchema,
+    TokenRefresh,
+)
 from security.interfaces import JWTAuthManagerInterface
+from security.passwords import hash_password, verify_password
+from security.utils import generate_secure_token
+from service import remove_token, get_jwt_manager
 
 router = APIRouter()
 
-# Write your code here
+
+@router.post("/register/", response_model=UserRead, status_code=201)
+async def register(user: CreateUser, db: AsyncSession = Depends(get_db)):
+    try:
+        db_user = await get_user_by_email(db, user.email)
+        if db_user:
+            raise HTTPException(
+                status_code=409,
+                detail=f"A user with this email {user.email} already exists.",
+            )
+        new_user = await create_user(db, user)
+        new_token = ActivationTokenModel(user_id=new_user.id)
+        db.add(new_token)
+        await db.commit()
+        await db.refresh(new_token)
+
+        return new_user
+
+    except HTTPException:
+        raise
+    except Exception:
+        raise HTTPException(
+            status_code=500, detail="An error occurred during user creation."
+        )
+
+
+@router.post("/activate/", response_model=dict)
+async def activate(model: TokenActivate, db: AsyncSession = Depends(get_db)):
+    user_status = await get_user_by_email(db, model.email)
+    if user_status.is_active:
+        raise HTTPException(
```

You call get_user_by_email and then immediately access `user_status.is_active` without checking whether the user exists. If the email doesn't exist this will raise an AttributeError instead of returning the expected 400/invalid response. Add a `if user_status is None: ...` guard and treat missing user as invalid input per the spec. This affects activation behavior when email is unknown.